tests/python_api/audio_unit_test.py

```diff
@@ -0,0 +1,176 @@
+# Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from amd.rocal.pipeline import pipeline_def
+from amd.rocal.plugin.pytorch import ROCALAudioIterator
+import amd.rocal.fn as fn
+import random
+import os
+import sys
+import matplotlib.pyplot as plt
+import torch
+import numpy as np
+import timeit
+from parse_config import parse_args
+
+np.set_printoptions(threshold=1000, edgeitems=10000)
+seed = random.SystemRandom().randint(0, 2**32 - 1)
+
+test_case_augmentation_map = {
+    0: "audio_decoder",
+}
+
+def plot_audio_wav(audio_tensor, idx):
+    # audio is expected as a tensor
+    audio_data = audio_tensor.detach().numpy()
+    audio_data = audio_data.flatten()
+    plt.plot(audio_data)
+    plt.savefig("OUTPUT_FOLDER/AUDIO_READER/" + str(idx) + ".png")
+    plt.close()
+
+def verify_output(audio_tensor, rocal_data_path, roi_tensor, test_results, case_name):
+    ref_path = f'{rocal_data_path}/rocal_data/GoldenOutputsTensor/reference_outputs_audio/{case_name}_output.bin'
+    data_array = np.fromfile(ref_path, dtype=np.float32)
+    audio_data = audio_tensor.detach().numpy().flatten()
+    roi_data = roi_tensor.detach().numpy()
+    matched_indices = 0
+    for j in range(roi_data[0]):
+        ref_val = data_array[j]
+        out_val = audio_data[j]
```

Comment:
Can we not check roi_data? Do you need audio_data?

Reply:
roi_data contains the roi values only
Audio data is the tensor with decoded/augmented audio